app/javascript/components/party/index.js

```diff
@@ -8,48 +8,46 @@ import Metrics from "./metrics";
 import Candidates from "./Candidates";
 
 export default class Party extends PureComponent {
+    constructor() {
+        super();
+
+        this.state = {
+            party: {},
+            candidates: []
+        }
+    }
+
+    componentDidMount() {
+      fetch("/parties/" + this.props.match.params.id + ".json")
+      .then(res => res.json())
+      .then(data =>
+        this.setState({
+          party: data,
+          candidates: data.candidates
```

Comment:
do we need to store this info? I'm saying this because we are already storing all the data in the `party` property.